### PR TITLE
Move initial key-exchange to infra

### PR DESCRIPTION
**Pull request by cloudnoize**
- do_key_exchange() is done as part of the with_bft_network decorator. by default it won't be done, unless one passes ```rotate_keys=True```

- ```test_rough_initial_key_exchange``` is marked as skip.

### CODE REVIEW

File: tests/apollo/test_skvbc_fast_path.py [hunk 1]
```diff
@@ -87,11 +87,17 @@ async def test_fast_to_slow_path_transition(self, bft_network, tracker):
 
         Finally the decorator verifies the KV execution.
         """
-        await bft_network.do_key_exchange()
         bft_network.start_all_replicas()
 
-        await bft_network.wait_for_fast_path_to_be_prevalent(
-            run_ops=lambda: tracker.run_concurrent_ops(num_ops=20, write_weight=1), threshold=20)
```
Comment (teoparvanov) [inline tests/apollo/test_skvbc_fast_path.py:94]: hmm, I think there's some issue with your rebase - you seem to have overridden the recent changes by Kostas?

**Review by teoparvanov — COMMENTED** [review]
Awesome, thanks for fixing this Eran!

File: tests/apollo/test_skvbc_fast_path.py (hunk 1 above)
Comment (cloudnoize) [reply tests/apollo/test_skvbc_fast_path.py:94]: mmm, maybe, anyway I didn't plan to merge this PR.
I'll create a new one and merge my code to recent code

File: tests/apollo/util/bft.py [hunk 2]
```diff
@@ -114,7 +114,7 @@ async def wrapper(*args, **kwargs):
                 nursery.cancel_scope.cancel()
     return wrapper
 
-def with_bft_network(start_replica_cmd, selected_configs=None, num_clients=None, num_ro_replicas=0):
+def with_bft_network(start_replica_cmd, selected_configs=None, num_clients=None, num_ro_replicas=0, rotate_keys=False):
```
Comment (teoparvanov) [inline tests/apollo/util/bft.py:117]: shouldn't there be changes in the tests too ?

Comment (cloudnoize) [reply tests/apollo/util/bft.py:117]: yeah, I'm adding them one by one

**Review by teoparvanov — APPROVED** [review]
LGTM!

File: tests/apollo/util/bft.py [hunk 3]
```diff
@@ -133,6 +133,8 @@ async def wrapper(*args, **kwargs):
                 bft_network = kwargs.pop("bft_network")
                 bft_network.is_existing = True
                 with log.start_task(action_type=async_fn.__name__):
+                    if rotate_keys:
+                        await bft_network.do_key_exchange()
```
Comment (teoparvanov) [inline tests/apollo/util/bft.py:137]: If the `bft_network` is externally provided (such as in Hermes), calling `do_key_exchange()` here probably doesn't make sense...

**Pull request merged**